### PR TITLE
Rework confusing shipping message on cart page when calculator is disabled and no methods are available.

### LINKED ISSUE #24674: Incorrect/misleading message for Shipping options in cart
**Describe the bug**
When setting up the Shipping options/calculations in the cart for a new WooCommerce store - you can select from the following two options:

Screenshot 2019-09-24 at 13 45 19

if you unselect both of those options and then go view the shipping options in the cart you get the following message:

Screenshot 2019-09-23 at 13 10 04

However, you cannot enter an address on the cart page and instead - based on the settings you have chosen that text should rather read something like this instead:

> Shipping costs are calculated during checkout

**To Reproduce**
1. Setup a new install of WooCommerce
2. Install Storefront
3. Uncheck both those options highlighted in the screenshot above
4. Add an item to your cart and view the message alongside the shipping options

**Expected behavior**

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by dhavalgshah**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Handles edge cases where users get message asking to _"Enter the address."_ on cart page when shipping calculator is disabled and no shipping methods are available. There is also another edge case where default customer location is set to no location by default and shipping zone for rest of the locations has no shipping methods.

These are sort of badly configured stores but because these related settings are on three different tabs this ought to happen.

Whereas it should convey that shipping can be updated on checkout as we don't have all the information to determine if shipping is actually available or not for the destination.

Closes #24674 .

### How to test the changes in this Pull Request:

1. Disable Shipping Calculator
2. Set shipping zone for other than your geolocation & disable any methods for _Locations not covered by your other zones_
3. Add products to cart & visit cart page
4. You should see _'Shipping options will be updated at checkout.'_

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix issue where if shipping calculator is disabled and no shipping methods are available then customer gets confusing message on cart page.

### CODE REVIEW

File: templates/cart/cart-shipping.php [hunk 1]
```diff
@@ -58,7 +58,11 @@
 			<?php endif; ?>
 			<?php
 		elseif ( ! $has_calculated_shipping || ! $formatted_destination ) :
-			echo wp_kses_post( apply_filters( 'woocommerce_shipping_may_be_available_html', __( 'Enter your address to view shipping options.', 'woocommerce' ) ) );
+			if ( is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' ) ) {
+				echo wp_kses_post( apply_filters( 'woocommerce_shipping_may_be_available_cart_nocalc_html', __( 'Shipping options will be updated at checkout.', 'woocommerce' ) ) );
```
Comment (claudiosanches) [inline templates/cart/cart-shipping.php]: Please update the message to:

```
Shipping costs are calculated during checkout
```

Also this `woocommerce_shipping_may_be_available_cart_nocalc_html` filter could get a more descriptive name, like `woocommerce_shipping_on_cart_not_enabled_html`.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Thanks for your help, there's a few small issues to fix before we can merge.

File: templates/cart/cart-shipping.php (hunk 1 above)
Comment (dhavalgshah) [reply templates/cart/cart-shipping.php]: I used **Shipping options ...** in line with #22467. Using **Shipping costs ...** would deviate from that messaging. 

If the current message is not suitable, then we may use 

> Shipping options are calculated during checkout



Wrt hook, `woocommerce_shipping_on_cart_not_enabled_html` would imply that shipping is not enabled on the cart.

However, here shipping is enabled, but the shipping calculator is not enabled, we can't detect shipping destination, and there aren't any shipping methods for this scenario set up by the shop manager. Thus, we have to wait until checkout to find the exact shipping destination. Then if we don't have any shipping methods for that we can say we absolutely have no shipping options for the customer.

Maybe `woocommerce_shipping_may_be_available_calc_not_enabled_html` would be better? This would also align with `woocommerce_shipping_may_be_available_html` hook which is right after it and is contextually related.

Also, maybe a rework on your suggestion `woocommerce_shipping_not_available_on_cart_html`.

Comment (claudiosanches) [reply templates/cart/cart-shipping.php]: > Wrt hook, woocommerce_shipping_on_cart_not_enabled_html would imply that shipping is not enabled on the cart.

That filter will only run when `is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' )` returns true, so how shipping can be enabled and still hit this filter?

> If the current message is not suitable, then we may use

As I said, the message I suggested will fit better.

> is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' )

Comment (dhavalgshah) [reply templates/cart/cart-shipping.php]: Updated the message in the latest commit.

> That filter will only run when `is_cart()` && `'no' === get_option( 'woocommerce_enable_shipping_calc' )` returns true, so how shipping can be enabled and still hit this filter?

`'no' === get_option( 'woocommerce_enable_shipping_calc' )` checks if **_Shipping Calculator_** is enabled or not on the cart. 

So shipping can be enabled but we can't match any shipping methods automatically because either we can't locate shipping destination (Geolocation may be disabled) or matched zone does not have methods for current packages. But because we are on cart and shipping calculator is disabled, we can not be sure about shipping destination until the customer enters it manually on the checkout page.

Bottom line: Shipping is enabled but we neither have matching shipping methods available nor we have accurate shipping destination available to flat out say _No shipping available for %s_

Comment (claudiosanches) [reply templates/cart/cart-shipping.php]: @dhavalgshah I still don't understand why we can't change the filter name as I requested. If shipping is disabled on cart, that `woocommerce_shipping_may_be_available_cart_nocalc_html` doesn't makes sense, because isn't "may be available", it's for sure unavailable, since `get_option( 'woocommerce_enable_shipping_calc' )` will returns `no`, so no shipping calculator will got rendered in the cart.

Let me know if you can update.
